### ISSUE TITLE
Link only one of the output libs?

### ISSUE THREAD

**Issue by wudisheng** [bazelbuild/rules_foreign_cc#768]
Hi,

I've searched and tried a bit but seems this is a relatively new topic.

I have several external projects (configure_make, cmake), each has multiple output libs (i.e. I specify multiple entries in `out_static_libs`). But for my Bazel target, I want to link toward only one of them each, and found it difficult.

An example (not my real example) would be `openssl`, how can I link only `libcrypto.a` (without `libssl.a`)? Without defining multiple `configure_make` targets (which takes times longer to build).

My real cases are for external projects that

- Output different libs for C and C++;
- Output different libs with a specific feature (e.g. threading / no-threading);
- Output multiple libs each for a specific feature (e.g. file / networking / etc.);

And I do need to use all of them for different Bazel targets (e.g. link only `libcrypto.a` for one target, but both for another).

**Comment by wudisheng** [comment]
To be clear, actually, all the outputs in `out_static_libs` can not be linked all together, because some of them define the same identifiers.

**Comment by jsharpe** [comment]
You need to use a `filegroup` with the output_group set to the name of the library you want to grab. You can then reference this in a `cc_import` target to allow you to link against a single library.

**Comment by wudisheng** [comment]
I've tried this way, but encountered some difficulties re: the include paths (i.e. the header files / dirs) in `CcInfo` associated with the foreign targets.

For now, my workaround is to use a simple rule to remove the `linking_context` from the `CcInfo` of the foreign targets, while keeping `compilation_context`, but this is not an ideal way since it removes too much. Perhaps I can traverse into `linking_context` to remove input files other than my desired one(s), but it is still coupling with (if any) `linkopts` or other information set by the foreign rules.

I filed this issue to explore if there's any official way of handling such demands, better adding a set of examples if so. :)
